rasa/core/training/loading.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import os
+from pathlib import Path
+from typing import Text, Optional, Dict, List, Union
+
+import rasa.utils.io as io_utils
+from rasa.core.domain import Domain
+from rasa.core.interpreter import NaturalLanguageInterpreter, RegexInterpreter
+from rasa.core.training.story_reader.story_reader import StoryReader
+from rasa.core.training.structures import StoryStep
+from rasa.data import YAML_FILE_EXTENSIONS, MARKDOWN_FILE_EXTENSION
+
+logger = logging.getLogger(__name__)
+
+
+def _get_reader(
+    filename: Text,
+    domain: Domain,
+    interpreter: NaturalLanguageInterpreter = RegexInterpreter(),
+    template_variables: Optional[Dict] = None,
+    use_e2e: bool = False,
+) -> StoryReader:
+    from rasa.core.training.story_reader import YAMLStoryReader
+    from rasa.core.training.story_reader import MarkdownStoryReader
+
+    if filename.endswith(MARKDOWN_FILE_EXTENSION):
+        return MarkdownStoryReader(
+            interpreter, domain, template_variables, use_e2e, filename
+        )
+    elif filename.split(".")[-1] in YAML_FILE_EXTENSIONS:
```

Comment:
how about this fancy boy https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffix ? 

or 
```suggestion
    elif any(f.endswith(extension) for extension in YAML_FILE_EXTENSIONS)
```

both seem a bit cleaner / more robust then magic string + magic number

Reply:
> how about this fancy boy 

That looks great. I'll do that in-place. I think the migration to the pathlib is a separate topic.